webhooks/virtualmachine/v1alpha2/validation/virtualmachine_validator.go

```diff
@@ -249,6 +249,19 @@ func (v validator) validateBootstrap(
 			allErrs = append(allErrs, field.TypeInvalid(p, vAppConfig,
 				"properties and rawProperties are mutually exclusive"))
 		}
+
+		if properties := vAppConfig.Properties; len(properties) != 0 {
+			for _, property := range properties {
+				if key := property.Key; key == "" {
+					allErrs = append(allErrs, field.Invalid(p.Child("properties").Child("key"), key,
+						"key is a required field in vAppConfig Properties"))
+				}
+				if value := property.Value; value.From != nil && value.Value != nil {
+					allErrs = append(allErrs, field.Invalid(p.Child("properties").Child("value"), value,
```

Comment:
Does using the `value` as the value cause the type to be a part of the error message? (We should avoid serializing types into the error strings)

Reply:
Yes it does. Makes sense. I am only putting `"value"` as part of error strings then.